pkg/virt-controller/watch/migration.go

```diff
@@ -133,33 +131,31 @@ func (md *MigrationController) execute(key string) error {
 		return nil
 	}
 
-	logger := log.Log.Object(obj.(*kubev1.Migration))
-	// Copy migration for future modifications
-	if obj, err = scheme.Scheme.Copy(obj.(runtime.Object)); err != nil {
-		logger.Reason(err).Error("could not copy migration object")
-		return err
-	}
 	migration := obj.(*kubev1.Migration)
 
-	vm, exists, err := md.vmService.FetchVM(migration.GetObjectMeta().GetNamespace(), migration.Spec.Selector.Name)
+	// Copy migration for future modifications
+	migrationCopy := migration.DeepCopy()
```

Comment:
why not just do this and get rid of migrationCopy?
```
migration = migration.DeepCopy()
```

having two migration objects around, one that can be mutated and another that we shouldn't, makes this function prone to errors in the future i think.

Reply:
@davidvossel We had it before the refactoring
```
if obj, err = scheme.Scheme.Copy(obj.(runtime.Object)); err != nil {
		logger.Reason(err).Error("could not copy migration object")
		return err
	}
	migration := obj.(*kubev1.Migration)
```
so I just wanted to save the same approach, I think we can review it again when we will have working migration :smile:

Reply:
yeah, this really doesn't matter. Migrations are being stripped out and this code isn't going to even exist (likely by the end of the day even)